pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -193,3 +239,106 @@ func (r *PullSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Named(controllers.PullSecretControllerName).
 		Complete(r)
 }
+
+func (r *PullSecretReconciler) parseRHRegistryKeys(ps *v1.Secret) (*serializedAuthMap, error) {
+	var pullSecretData *serializedAuthMap
+	if data := ps.Data[v1.DockerConfigJsonKey]; len(data) > 0 {
+		if err := json.Unmarshal(data, &pullSecretData); err != nil {
+			return nil, err
+		}
+	}
+	return pullSecretData, nil
+}
+
+// checkRHRegistryKeys checks whether the rhRegistry keys:
+//   - redhat.registry.io
+//   - registry.connect.redhat.com"
+// are present in the pullSecret
+func (r *PullSecretReconciler) checkRHRegistryKeys(psData *serializedAuthMap) (foundKeys []string) {
+	rhKeys := []string{
+		"registry.redhat.io",
+		"registry.connect.redhat.com",
+	}
+	foundKeys = make([]string, 0, len(rhKeys))
+
+	if psData == nil {
+		return foundKeys
+	}
+
+	for _, key := range rhKeys {
+		if auth, ok := psData.Auths[key]; ok && len(auth.Auth) > 0 {
+			r.log.Infof("Found token: %s\n", key)
+			foundKeys = append(foundKeys, key)
+		}
+	}
+
+	return foundKeys
+}
+
+func (r *PullSecretReconciler) updateRHKeysCondition(foundKeys []string) *status.Condition {
+	cond := &status.Condition{
+		Type:    aro.RedHatKeyPresent,
+		Status:  v1.ConditionFalse,
+		Message: "No Red Hat registry keys found in pull-secret.",
+		Reason:  "CheckDone",
+	}
+
+	if len(foundKeys) == 0 {
+		return cond
+	}
+
+	sb := strings.Builder{}
+	sb.WriteString("Red Hat registry keys present in pull-secret: ")
+	for _, key := range foundKeys {
+		sb.WriteString(key + ", ")
+	}
+
+	cond.Status = v1.ConditionTrue
+	cond.Message = sb.String()
+
+	return cond
+}
+
+// switchSamples enables/disables the samples if there's no appropriate pull secret
+func (r *PullSecretReconciler) switchSamples(ctx context.Context, enable bool) (bool, error) {
+	updated := false
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		c, err := r.samplescli.SamplesV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if c.Spec.SamplesRegistry != "" {
+			// if the samples registry point elsewhere no action
+			return nil
+		}
+
+		if enable {
+			if c.Spec.ManagementState == operatorv1.Managed {
+				return nil
+			}
+			c.Spec.ManagementState = operatorv1.Managed
+		} else {
+			if c.Spec.ManagementState == operatorv1.Removed {
+				return nil
+			}
+			c.Spec.ManagementState = operatorv1.Removed
+		}
```

Comment:
Can we simplify this too somehow?
One possible way:
```
if (enabled && c.Spec.ManagementState == operatorv1.Managed ) || (!enabled && c.Spec.ManagementState == operatorv1.Removed)  {
    return
}
```
and next if should be simpler.
Goal would be to make it easier to read.